modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataMediator.java

```diff
@@ -121,41 +120,55 @@ public MetadataMediator(T componentModel) {
   public MetadataResult<ComponentMetadataDescriptor<T>> getMetadata(MetadataContext context, MetadataKey key) {
     try {
       Object resolvedKey = keyIdObjectResolver.resolve(key);
-      return getMetadata(context, p -> resolvedKey, MetadataAttributes.builder().withKey(key));
+      return getMetadata(context, resolvedKey, MetadataAttributes.builder().withKey(key));
     } catch (MetadataResolvingException e) {
       return failure(newFailure(e).onComponent());
     }
   }
 
+  /**
+   * Resolves the {@link ComponentMetadataDescriptor} for the associated {@code context} using the component
+   * {@link ParameterValueResolver} to resolve an object with the value of the {@link MetadataKey}.
+   *
+   * @param context current {@link MetadataContext} that will be used by the metadata resolvers.
+   * @return Successful {@link MetadataResult} if the MetadataTypes are resolved without errors Failure {@link MetadataResult}
+   *         when the Metadata retrieval of any element fails for any reason
+   */
   public MetadataResult<ComponentMetadataDescriptor<T>> getMetadata(MetadataContext context,
                                                                     ParameterValueResolver metadataKeyResolver) {
-    return getMetadata(context, metadataKeyResolver, MetadataAttributes.builder());
+    try {
+      Object keyValue;
+      MetadataResult keyValueResult = getMetadataKeyObjectValue(metadataKeyResolver);
+      if (!keyValueResult.isSuccess()) {
+        return keyValueResult;
+      } else {
+        keyValue = keyValueResult.get();
+      }
+
+      MetadataAttributes.MetadataAttributesBuilder builder = MetadataAttributes.builder();
+      if (keyValue != null) {
+        builder.withKey(keyIdObjectResolver.reconstructKeyFromType(keyValue));
+      }
+
+      return getMetadata(context, keyValue, builder);
```

Comment:
is the `null` key being propagated all the way to the Typeresolver the developer implemented?